software/py/vanilla_stats_parser.py

```diff
@@ -0,0 +1,640 @@
+    self.__print_manycore_stats_tile_group_timing(manycore_stats_file)
+    self.__print_manycore_stats_tile_timing(manycore_stats_file)
+    self.__print_manycore_stats_miss(manycore_stats_file)
+    self.__print_manycore_stats_stall(manycore_stats_file)
+    self.__print_manycore_stats_instr(manycore_stats_file)
+    manycore_stats_file.close()
+
+  # prints all four types of stats, timing, instruction,
+  # miss and stall for each tile group in a separate file  
+  def print_per_tile_group_stats_all(self):
+    stats_path = os.getcwd() + "/stats/tile_group/"
+    if not os.path.exists(stats_path):
+      os.mkdir(stats_path)
+    for tg_id in range(self.num_tile_groups):
+      stat_file = open( (stats_path + "tile_group_" + str(tg_id) + "_stats.log"), "w")
+      self.__print_per_tile_group_stats_timing(tg_id, stat_file)
+      self.__print_per_tile_group_stats_miss(tg_id, stat_file)
+      self.__print_per_tile_group_stats_stall(tg_id, stat_file)
+      self.__print_per_tile_group_stats_instr(tg_id, stat_file)
+      stat_file.close()
+
+
+
+  # prints all four types of stats, timing, instruction,
+  # miss and stall for each tile in a separate file  
+  def print_per_tile_stats_all(self):
+    stats_path = os.getcwd() + "/stats/tile/"
+    if not os.path.exists(stats_path):
+      os.mkdir(stats_path)
+    for y in range(self.manycore_dim_y):
+      for x in range(self.manycore_dim_x):
+        stat_file = open( (stats_path + "tile_" + str(y) + "_" + str(x) + "_stats.log"), "w")
+        self.__print_per_tile_stats_timing(y, x, stat_file)
+        self.__print_per_tile_stats_miss(y, x, stat_file)
+        self.__print_per_tile_stats_stall(y, x, stat_file)
+        self.__print_per_tile_stats_instr(y, x, stat_file)
+        stat_file.close()
+
+
+  # main public method
+  # default stats generator
+  def generate_stats(self, input_file):
+    self.traces = []
+    with open(input_file) as f:
+      csv_reader = csv.DictReader (f, delimiter=",")
+
+      # Generate list of instruction/miss/stall types
+      header = csv_reader.fieldnames
+      for item in header:
+        if (item.startswith('instr_')):
+          if (not item == 'instr_total'):
+            self.instr_list += [item]
+        elif (item.startswith('miss_')):
+          self.miss_list += [item]
+        elif (item.startswith('stall_')):
+          self.stalls_list += [item]
+        else:
+          self.stats_list += [item]
+      self.all_ops_list = self.stats_list + self.instr_list + self.miss_list + self.stalls_list
+
+      for row in csv_reader:
+        trace = {}
+        for stat in self.stats_list:
+          trace[stat] = int(row[stat])
+        for instr in self.instr_list:
+          trace[instr] = int(row[instr])
+        for stall in self.stalls_list:
+          trace[stall] = int(row[stall])
+        for miss in self.miss_list:
+          trace[miss] = int(row[miss])
+        self.traces.append(trace)
```

Comment:
Can you explain what's going on here?

Reply:
It's reading the trace for every type of operations (instruction, miss, stall, stats) and putting them into a trace line. But I made it shorter, how does it look now?